### PR TITLE
feat(license-service): map status code from verifyPkpass API

### PR DESCRIPTION
**Pull request by osk**
SmartSolution API now returns an error code based on the result of verifying a pkpass license, pass that data along to the client.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by thordurhhh — APPROVED** [review]

File: libs/api/domains/license-service/src/lib/client/driving-license-client/drivingLicenseService.api.ts [hunk 1]
```diff
@@ -481,8 +490,14 @@ export class GenericDrivingLicenseApi
...
+      // Is there a status code from the service?
+      const serviceErrorStatus = result.error.serviceError?.status
+
+      // Use status code, or http status code from serivce, or "0" for unknown
+      const status = serviceErrorStatus ?? (result.error.statusCode || 0)
+
       error = {
-        status: (result.error.statusCode || 0).toString(),
+        status: status.toString(),
         message: result.error.serviceError?.message ?? 'Unknown error',
```
Comment (maranomynet) [inline libs/api/domains/license-service/src/lib/client/driving-license-client/drivingLicenseService.api.ts:501]:
```suggestion
        message: result.error.serviceError?.message || 'Unknown error',
```

Use fallback error message for non-nully-yet-empty messges?

Comment (maranomynet) [inline libs/api/domains/license-service/src/lib/client/driving-license-client/drivingLicenseService.api.ts:501]:
```suggestion
        message: result.error.serviceError?.message || 'Unknown error',
```

Use fallback error message for non-nully-yet-empty messges?

File: libs/api/domains/license-service/src/lib/licenceService.type.ts [hunk 2]
```diff
@@ -102,8 +102,14 @@ export type GenericUserLicense = {
 }
 
 export type PkPassVerificationError = {
+  // Generic placeholder for a status code, could be the HTTP status code, code
+  // from API, or empty string. Semantics need to be defined per license type
```
Comment (maranomynet) [inline libs/api/domains/license-service/src/lib/licenceService.type.ts:106]:
```suggestion
  /**
   * Generic placeholder for a status code, could be the HTTP status code, code
   * from API, or empty string. Semantics need to be defined per license type
   */
```

File: libs/api/domains/license-service/src/lib/licenceService.type.ts [hunk 3]
```diff
@@ -102,8 +102,14 @@ export type GenericUserLicense = {
...
   status: string
+
+  // Generic placeholder for a status message, from API, or empty "Unknown error".
+  // Semantics need to be defined per license type
```
Comment (maranomynet) [inline libs/api/domains/license-service/src/lib/licenceService.type.ts:110]:
```suggestion
  /**
   * Generic placeholder for a status message, from API, or empty "Unknown error".
   * Semantics need to be defined per license type
   */
```

File: libs/api/domains/license-service/src/lib/licenceService.type.ts [hunk 4]
```diff
@@ -102,8 +102,14 @@ export type GenericUserLicense = {
...
   message: string
+
   // data is used to pass along the error from originator, e.g. SmartSolution
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (maranomynet) [inline libs/api/domains/license-service/src/lib/licenceService.type.ts:114]:
```suggestion
  /**
   * data is used to pass along the error from originator, e.g. SmartSolution
   * eslint-disable-next-line @typescript-eslint/no-explicit-any
   */
```

File: libs/api/domains/license-service/src/lib/licenceService.type.ts [hunk 5]
```diff
@@ -102,8 +102,14 @@ export type GenericUserLicense = {
...
   data?: any
```
Comment (maranomynet) [inline libs/api/domains/license-service/src/lib/licenceService.type.ts:115]:
```suggestion
  data?: unknown
```

Avoid `any`, no?

**Review by maranomynet — DISMISSED** [review]

File: libs/api/domains/license-service/src/lib/client/driving-license-client/drivingLicenseService.api.ts (hunk 1 above)
Comment (osk) [reply libs/api/domains/license-service/src/lib/client/driving-license-client/drivingLicenseService.api.ts:501]: My thinking was that an empty message was a message in itself.. but maybe this makes more sense

**Review by maranomynet — APPROVED** [review]

**Pull request merged**